### ISSUE TITLE
Distributed query fails with allow_experimental_analyzer and array join 

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#64755]
https://fiddle.clickhouse.com/80cfeb73-a252-4032-9a52-c8734d7e8dd0

```sql
create table hourly(
  hour datetime,
  `metric.names` Array(String),
  `metric.values` Array(Int64)
) Engine=Memory
as select '2020-01-01', ['a', 'b'], [1,2];

set allow_experimental_analyzer=0;

SELECT
     toDate(hour) AS day,
     `metric.names`,
     sum(`metric.values`)
FROM remote('127.0.0.{1,2}', default, hourly)
ARRAY JOIN metric
GROUP BY
     day,
     `metric.names`
format PrettyCompactMonoBlock;

   +--------day-+-metric.names-+-sum(metric.values)-+
1. | 2020-01-01 | b            |                  4 |
2. | 2020-01-01 | a            |                  2 |
   +------------+--------------+--------------------+

set allow_experimental_analyzer=1;

SELECT
     toDate(hour) AS day,
     `metric.names`,
     sum(`metric.values`) 
FROM default.hourly
ARRAY JOIN metric
GROUP BY
     day,
     `metric.names`
format PrettyCompactMonoBlock;

   +--------day-+-metric.names-+-sum(metric.values)-+
1. | 2020-01-01 | b            |                  2 |
2. | 2020-01-01 | a            |                  1 |
   +------------+--------------+--------------------+


SELECT
     toDate(hour) AS day,
     `metric.names`,
     sum(`metric.values`)
FROM remote('127.0.0.{1,2}', default, hourly)
ARRAY JOIN metric
GROUP BY
     day,
     `metric.names`
format PrettyCompactMonoBlock;

DB::Exception: Cannot find column `tupleElement(__table1.metric, 'names'_String)
```

**Comment by santileira** [comment]
@KochetovNicolai We have a similar use case and after upgrading to `24.3.12.75`, we are still having the same issue. Do you know why?

cc: @vitaliyf
Example:
```
CREATE TABLE default.crash_group
(
    `date` Date,
    `app_id` LowCardinality(String),
    `prop.key` Array(String),
    `prop.value` Array(String)
)
ENGINE = Distributed('{cluster}', '', 'crash_group_local')


CREATE TABLE default.crash_group_local
(
    `date` Date,
    `app_id` LowCardinality(String),
    `prop.key` Array(String),
    `prop.value` Array(String)
)
ENGINE = ReplicatedReplacingMergeTree('/clickhouse/{cluster}/default/tables/default.crash_group_local/20220617/{shard}', '{replica}')
PARTITION BY date
ORDER BY (date, app_id)
TTL ttl_date
SETTINGS index_granularity = 8192

-- working
SELECT prop.key AS prop_key
FROM crash_group
ARRAY JOIN prop
WHERE app_id = '<app_id>' AND date >= '2021-03-03' AND date <= '2021-03-04'
GROUP BY prop_key
LIMIT 100
SETTINGS allow_experimental_analyzer = 0;

Ok

-- not working
SELECT prop.key AS prop_key
FROM crash_group
ARRAY JOIN prop
WHERE app_id = '<app_id>' AND date >= '2021-03-03' AND date <= '2021-03-04'
GROUP BY prop_key
LIMIT 100
SETTINGS allow_experimental_analyzer = 1;

Received exception from server (version 24.3.12):
Code: 8. DB::Exception: Received from ch-99-a..... DB::Exception: Cannot find column `tupleElement(__table1.prop, 'key'_String)` in source stream, there are only columns: [tupleElement(__table1.__array_join_expression_1, 'key'_String)]. (THERE_IS_NO_COLUMN)

-- working
SELECT prop.key AS prop_key
FROM crash_group
ARRAY JOIN prop as p
WHERE app_id = '<app_id>' AND date >= '2021-03-03' AND date <= '2021-03-04'
GROUP BY prop_key
LIMIT 100
SETTINGS allow_experimental_analyzer = 1;

Ok
```

**Comment by den-crane** [comment]
@santileira it was never backported into 24.3


```
cd clickhouse/docs/changelogs
grep 64801 *

v24.5.5.41-stable.md:* Backported in [#65621](https://github.com/ClickHouse/ClickHouse/issues/65621): Fix `Cannot find column` in distributed query with `ARRAY JOIN` by `Nested` column. Fixes [#64755](https://github.com/ClickHouse/ClickHouse/issues/64755). [#64801](https://github.com/ClickHouse/ClickHouse/pull/64801) ([Nikolai Kochetov](https://github.com/KochetovNicolai)).

v24.5.5.78-stable.md:* Backported in [#65621](https://github.com/ClickHouse/ClickHouse/issues/65621): Fix `Cannot find column` in distributed query with `ARRAY JOIN` by `Nested` column. Fixes [#64755](https://github.com/ClickHouse/ClickHouse/issues/64755). [#64801](https://github.com/ClickHouse/ClickHouse/pull/64801) ([Nikolai Kochetov](https://github.com/KochetovNicolai)).

v24.6.1.4423-stable.md:* Fix `Cannot find column` in distributed query with `ARRAY JOIN` by `Nested` column. Fixes [#64755](https://github.com/ClickHouse/ClickHouse/issues/64755). [#64801](https://github.com/ClickHouse/ClickHouse/pull/64801) ([Nikolai Kochetov](https://github.com/KochetovNicolai)).
```

**Comment by vitaliyf** [comment]
@den-crane Thanks - there was https://github.com/ClickHouse/ClickHouse/pull/65619 but did it get closed without merging to 24.3?

**Comment by den-crane** [comment]
> @den-crane Thanks - there was #65619 but did it get closed without merging to 24.3?

right
